crates/typst-pdf/src/gradient.rs

```diff
@@ -145,10 +146,19 @@ pub fn write_gradients(
                         .bits_per_component(16)
                         .bits_per_flag(8)
                         .shading_type(StreamShadingType::CoonsPatch)
-                        .decode([
-                            0.0, 1.0, 0.0, 1.0, range[0], range[1], range[2], range[3],
-                            range[4], range[5],
-                        ])
+                        .decode(match color_space {
+                            ColorSpace::Cmyk => SmallVec::<[f32; 12]>::from_buf([
+                                0.0, 1.0, 0.0, 1.0, range[0], range[1], range[2],
+                                range[3], range[4], range[5], range[6], range[7],
+                            ]),
+                            ColorSpace::D65Gray => {
+                                smallvec![0.0, 1.0, 0.0, 1.0, range[0], range[1]]
+                            }
+                            _ => smallvec![
+                                0.0, 1.0, 0.0, 1.0, range[0], range[1], range[2],
+                                range[3], range[4], range[5],
+                            ],
+                        })
```

Comment:
`decode` takes an `IntoIterator<Item = f32>`, so you can simplify this a lot by just appending the range, which already only contains the correct number of components:
```rs
.decode([0.0, 1.0, 0.0, 1.0].into_iter().chain(range))
```

Reply:
That's a very nice simplification indeed! And I also like that, as for your previous suggestion, it allows in principle to add other color paces in the future without changing this part of the code. Thanks!